runtime/lua/vim/lsp/log.lua

```diff
@@ -32,12 +32,12 @@ local function notify(msg, level)
   end
 end
 
-local logfilename = vim.fs.joinpath(vim.fn.stdpath('log'), 'lsp.log')
+local logfilename = vim.fs.joinpath(vim.fn.stdpath('log') --[[@as string]], 'lsp.log')
```

Comment:
does LuaLs support parameterizing the return type based on the input?

alternatively, maybe we should just fib a little and define `stdpath():string` in the meta files. Almost no one ever uses the `_dirs` cases.

Reply:
I believe Lua's type annotations aren't powerful enough for that yet. Maybe we could use overloads but the type for this one is autogenerated and idk if we have support for that yet (but @lewis6991 would know better).

Reply:
Return type polymorphism is generally difficult for a type system, so most don't allow it. The only exception is generics.

Reply:
Let's define stdpath as returning `string` then